types/pixi/shaders/abstractBaseShader.d.ts

```diff
@@ -0,0 +1,100 @@
+/**
+ * This class defines an interface which all shaders utilize
+ */
+declare abstract class AbstractBaseShader extends PIXI.Shader {
+  constructor(program: PIXI.Program, uniforms: AbstractBaseShader.Uniforms);
+
+  /**
+   * The initial default values of shader uniforms
+   */
+  protected _defaults: AbstractBaseShader.Uniforms;
+
+  /**
+   * The default vertex shader used by all instances of AbstractBaseShader
+   */
+  static vertexShader: string;
+
+  /**
+   * The fragment shader which renders this source.
+   * A subclass of AbstractBaseShader must implement the fragmentShader static field.
+   */
+  static fragmentShader: string;
+
+  /**
+   * The default uniform values for the shader.
+   * A subclass of AbstractBaseShader must implement the defaultUniforms static field.
+   * @defaultValue `{}`
+   */
+  static defaultUniforms: AbstractBaseShader.Uniforms;
```

Comment:
It might be worth it to add an additional `@remarks` tag here and explicitly state that those are abstract.

Reply:
yeah, makes sense. I will add `@remarks`.

Reply:
Done